### PR TITLE
add key value http logger

### PR DESCRIPTION
**Pull request by nsmolenskii**
## Description

Add support key value http formatters

## Motivation and Context

Some log collecting software (like [Splunk](https://splunk.com/)), use key value logging formats by default.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

⚠️ depends on #357

### CODE REVIEW

File: logbook-core/src/main/java/org/zalando/logbook/KeyValueHttpLogFormatter.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class KeyValueHttpLogFormatter extends AbstractPreparedHttpLogFormatter {
+
+    @Override
+    public String format(final Map<String, Object> content) {
+        return content.entrySet().stream()
+                .map(entry -> String.format("%s=%s", entry.getKey(), entry.getValue()))
```
Comment (whiskeysierra) [inline logbook-core/src/main/java/org/zalando/logbook/KeyValueHttpLogFormatter.java]: Any escaping that needs to be done for values that contain a `=` e.g. the URL?

File: logbook-core/src/main/java/org/zalando/logbook/KeyValueHttpLogFormatter.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.zalando.logbook;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class KeyValueHttpLogFormatter extends AbstractPreparedHttpLogFormatter {
```
[the remark this answers is not in the document]
Comment (nsmolenskii) [reply logbook-core/src/main/java/org/zalando/logbook/KeyValueHttpLogFormatter.java]: Well, splunk just uses key-value pairs, but it does not mean that every key-value format consumer is splunk.

[the remark this answers is not in the document]
Comment (whiskeysierra) [reply logbook-core/src/main/java/org/zalando/logbook/KeyValueHttpLogFormatter.java]: True, but until someone comes along who needs something similar outside of splunk I'd prefer to be specific. That gives us a bit more freedom in evolving this along with splunk, if needed.

File: logbook-core/src/main/java/org/zalando/logbook/KeyValueHttpLogFormatter.java (hunk 1 above)
Comment (nsmolenskii) [reply logbook-core/src/main/java/org/zalando/logbook/KeyValueHttpLogFormatter.java]: Well, that is a idiomatic question.
Splunk for example is eating such format like `uri=http://host?k=v` and allows you to search it. But also during the syntax highlighting, it suggest two pairs `uri=http://host` and `k=v`, which allow you to search those parameters bound to each other (full search) or independently (partial search). It is quite useful, when you have many parameters in uri, but you is interested in only some.

**Comment by whiskeysierra** [comment]
Can you resolve the conflict?

**Comment by whiskeysierra** [comment]
> Can you resolve the conflict?

Nevermind. I just saw that Github has a web editor for this now. Let's see whether it worked.

**Comment by whiskeysierra** [comment]
Not really. Looks like the whole README is now somehow modified.

**Comment by nsmolenskii** [comment]
> Not really. Looks like the whole README is now somehow modified.

Yeah 😞.  I had rebase my branch against master - looks better now.

**Comment by whiskeysierra** [comment]
Thanks a lot for the contribution! :tada: I'll prepare a new release as soon as I get to the office.

**Pull request merged**